projects/RabbitMQ.Client/util/NetworkOrderDeserializer.cs

```diff
@@ -43,12 +44,38 @@ internal static ushort ReadUInt16(ReadOnlySpan<byte> span)
             return BinaryPrimitives.ReadUInt16BigEndian(span);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static ushort ReadUInt16(ReadOnlySequence<byte> buffer)
+        {
+            if (!BinaryPrimitives.TryReadUInt16BigEndian(buffer.First.Span, out ushort value))
+            {
+                Span<byte> bytes = stackalloc byte[2];
+                buffer.Slice(0, 2).CopyTo(bytes);
+                value = BinaryPrimitives.ReadUInt16BigEndian(bytes);
```

Comment:
Maybe I miss something. Doesn't this under the covers do `MemoryMarshal.Read` which should copy into the ushort? Why is the stackalloc needed?

Reply:
I'm guessing it's due to the fact that `buffer` is a `ReadOnlySequence<byte>` vs what `BinaryPrimitives.ReadUInt16BigEndian' expects (`ReadOnlySpan<byte>`).

I'm not sure why `TryReadUInt16BigEndian` is needed. Since we check that we have enough data to read the `UInt16` value it should always succed.

Reply:
TryRead is faster if there is enough space in the first Span in the Sequence to read the data. If we do not have enough space, we stackalloc enough space (to save heap allocations) and copy the data in there to read without trying.

And yes, the `BinaryPrimitives` methods don't have overloads to read from a `ReadOnlySequence` :(